### PR TITLE
replica: swallow barrier exceptions in table_sync_and_check

### PR DESCRIPTION
**Pull request by Deexie**
table_sync_and_check triggers read barrier to synchronize schema and check whether a given table exists. If the barrier fails, e.g. due to timeout, barrier's exception is propagated.

Swallow exception and fall back on local schema check in case of barrier failure.

Fixes: #18490.

- [ ] ** Backport reason (please explain below if this patch should be backported or not) **
No backport needed.

### CODE REVIEW

File: repair/table_check.cc [hunk 1]
```diff
@@ -13,17 +13,21 @@
...
+            auto sub = mm.get_abort_source().subscribe([&as] () noexcept {
+                if (!as.abort_requested()) {
+                    as.request_abort();
+                }
+            });
+
+            // Trigger read barrier to synchronize schema.
+            co_await mm.get_group0_barrier().trigger(as);
+        }
+    } catch (...) {
```
Comment (denesb) [inline repair/table_check.cc:29]: We don't want to swallow any exception here, just some specific exceptions, like the barrier timing out.
For this reason, we also shouldn't enclose all this code in `try..catch`, just the barrier itself.

**Comment by Deexie** [comment]
- swallow only `service::raft_operation_timeout_error`

**Comment by tgrabiec** [comment]
The barrier is there for correctness, we want repair to see all previous group0 changes, so we cannot ignore its failure.

If we don't see previous changes, repair may think table is dropped while it's actually not yet created.

There are other reasons mentioned here: https://github.com/scylladb/scylladb/issues/17658

**Comment by denesb** [comment]
> The barrier is there for correctness, we want repair to see all previous group0 changes, so we cannot ignore its failure.

So should we let the barrier's exception fail repair, or should we do something else here (retry)?

**Comment by tgrabiec** [comment]
> > The barrier is there for correctness, we want repair to see all previous group0 changes, so we cannot ignore its failure.
> 
> So should we let the barrier's exception fail repair, or should we do something else here (retry)?

We can start by failing repair, but the barrier should have a generous timeout so that it doesn't fail just because group0 is slow to catch up.

**Pull request closed without merging**